Allow nary choice in rewrite_strat (#14989)
coq/coq: doc/sphinx/addendum/generalized-rewriting.rst

   | eval @red_expr
   | fold @one_term
   | ( @rewstrategy )

>    | choice {+ @rewstrategy }

Reviewer: Would you move this line to the same spot as the previous "choice"?  doc_grammar is not always able to preserve ordering info that's obvious to humans.

Reviewer: Also, would you update this text 30-40 lines below to reflect the grammar change?
`left_biased_choice` (ugh) could be something like "use the first successful strategy".

```
:n:`choice @rewstrategy @rewstrategy`
   left_biased_choice
```

``progress`` tests progress of the argument :n:`@rewstrategy` and fails if no
progress was made, while ``try`` always succeeds, catching failures.

> ``choice`` is left-biased: it will launch the first strategy and fall back

Reviewer: ```suggestion
``choice`` uses the first successful strategy in the list of @rewstrategy.
```
and drop "on the next one in case of failure" on the next line.
